bitcoin/src/address.rs

```diff
@@ -727,18 +727,15 @@ impl NetworkValidation for NetworkUnchecked {
 /// * [BIP341 - Taproot: SegWit version 1 spending rules](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki)
 /// * [BIP350 - Bech32m format for v1+ witness addresses](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki)
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Address<V = NetworkChecked>
-where
-    V: NetworkValidation,
-{
-    /// The type of the address.
-    pub payload: Payload,
-    /// The network on which this address is usable.
-    pub network: Network,
-    /// Marker of the status of network validation.
-    _validation: PhantomData<V>,
+struct AddressInner {
+    payload: Payload,
+    network: Network,
 }
 
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
This lost documentation. Also we should document that `repr(transparent)` is internal and users shouldn't depend on it.